src/amoc_controller.erl

```diff
@@ -231,6 +254,38 @@ init_scenario(Scenario) ->
             skip
     end.
 
+-spec maybe_start_scenario_checking(amoc:scenario()) -> ok | skip.
+maybe_start_scenario_checking(Scenario) ->
+    Callbacks = [{continue, 0}, {terminate, 1}],
+    case are_callbacks_exported(Scenario, Callbacks) of
+        true ->
+            schedule_scenario_checking(Scenario, validation_interval()),
+            lager:info("validation has been started", []),
+            ok;
+        false ->
+            lager:info("validation is not started as callbacks=~p are not"
+                       " exported", [Callbacks]),
+            skip
+    end.
+
+-spec are_callbacks_exported(amoc:scenario(), proplists:proplist()) -> boolean().
+are_callbacks_exported(Scenario, Callbacks) ->
+    case code:ensure_loaded(Scenario) of
+        {module, Scenario} ->
+            lists:all(fun({Func, Arity}) ->
+                              erlang:function_exported(Scenario, Func, Arity)
+                      end, Callbacks);
+        Error ->
+            lager:error("scenario module ~p cannot be found, reason: ~p",
+                        [Scenario, Error]),
+            false
+    end.
+
+-spec schedule_scenario_checking(amoc:scenario(), non_neg_integer()) ->
+    reference().
+schedule_scenario_checking(Scenario, Interval) ->
+    erlang:send_after(Interval, ?SERVER, {check, Scenario, Interval}).
```

Comment:
I don't trust loops done this way, I was recently bitten in the backside by such code. Do we know what is going to happen if `Scenario:continue()` fails? Will the whole test crash? Will it terminate gracefully? Or will it keep running forever, without any more checking?

Reply:
>I don't trust loops done this way, I was recently bitten in the backside by such code.
@bartekgorny what do you mean here?

I'm guessing that `amoc_controller` would crash and be restarted by the supervisor and test would last forever as checking wouldn't work after restart. I did not catch exceptions when calling the callbacks as I wanted to stick to the rest of the code. What do you suggests then?